src/authutils/token/fastapi.py

```diff
@@ -0,0 +1,105 @@
+from asyncio import Future, get_event_loop
+from collections import OrderedDict
+
+import httpx
+from fastapi import Security, HTTPException
+from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
+from starlette.status import HTTP_403_FORBIDDEN
+
+from . import core
+from ..errors import JWTError, AuthError
+
+bearer = HTTPBearer()
+
+_jwt_public_keys = {}
+
+
+def access_token(*audiences, issuer=None, allowed_issuers=None, purpose=None):
+    """
+    Validate and return the JWT bearer token in HTTP header::
+
+        from fastapi import Depends
+
+        @app.get("/whoami")
+        def whoami(token=Depends(access_token("user", "openapi", purpose="access"))):
+            return token["iss"]
+
+    :param audiences: Required, all must occur in ``aud``.
```

Comment:
nitpick: can we use consistent docstring format? I prefer the format used https://github.com/uc-cdis/authutils/pull/40/files#diff-5a39ab7f8c2ab7243a2e4686c27548aaR63 I think that's the one that's recommended in our style guid